crates/router/src/core/payments/operations/payment_confirm.rs

```diff
@@ -555,7 +555,27 @@ impl<F: Clone + Send, Ctx: PaymentMethodRetrieve> Domain<F, api::PaymentsRequest
         requeue: bool,
         schedule_time: Option<time::PrimitiveDateTime>,
     ) -> CustomResult<(), errors::ApiErrorResponse> {
-        helpers::add_domain_task_to_pt(self, state, payment_attempt, requeue, schedule_time).await
+        // This spawns this futures in a background thread, the exception inside this future won't affect
+        // the current thread and the lifecycle of spawn thread is not handled by runtime.
+        // So when server shutdown won't wait for this thread's completion.
+        let m_payment_attempt = payment_attempt.clone();
+        let m_state = state.clone();
+        let m_self = *self;
+        tokio::spawn(
+            async move {
+                helpers::add_domain_task_to_pt(
+                    &m_self,
+                    m_state.as_ref(),
+                    &m_payment_attempt,
+                    requeue,
+                    schedule_time,
+                )
+                .await
+            }
+            .in_current_span(),
```

Comment:
Does this also propagate the fields captured in the current span? how the log structure will be done on the log aggregator?

Reply:
the log structure will be same and it can capture the fields in current span.

```
{
    "message": "[GENERIC_INSERT - EVENT] diesel_models::query::generics",
    "pid": 8,
    "level": "DEBUG",
    "target": "diesel_models::query::generics",
    "service": "router",
    "line": 88,
    "file": "hyperswitch_oss/crates/diesel_models/src/query/generics.rs",
    "fn": "generic_insert",
    "full_name": "diesel_models::query::generics::generic_insert",
    "flow": "PaymentsConfirm",
    "request_method": "POST",
    "request_url_path": "/payments/***/confirm",
    "request_id": "***",
    "merchant_id": "***",
    "extra": {
        ...
    }
}
```

The *** are captured values